File: SoftGallery/SoftGallery.Dominio/Models/Pedido.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace SoftGallery.Dominio.Models
{
    public class Pedido
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public string Id { get; set; }
        public Cliente Cliente { get; set; }
        public List<Produto> Produtos { get; set; } = new List<Produto>();
        public DateTime Data { get; set; }
        public decimal ValorTotal
        {
            get
            {
                return this.Produtos.Sum(p => p.Preco);
            }
        }

        public Pedido(List<Produto> produtos, Cliente cliente, DateTime data)
        {
            this.Produtos = produtos;
            this.Cliente = cliente;
            this.Data = data;
        }

        private Pedido() { }
    }
}

File: SoftGallery.Web/src/types/pedido.ts
export interface PedidoDTO {
  produtosIds: string[];
}

export interface Produto {
  id: string;
  nome: string;
  preco: number;
}

export interface Cliente {
  id: string;
  nome: string;
  email: string;
}

export interface Pedido {
  id: string;
  dataPedido: string;
  cliente: Cliente;
  produtos: Produto[];
}
